pkg/flexvolume/flexvolume.go

```diff
@@ -143,12 +150,11 @@ func processOpts(optsStr string) (Options, error) {
 
 // ExecDriver executes the appropriate FlexvolumeDriver command based on
 // recieved call-out.
-func ExecDriver(driver Driver, args []string) {
+func ExecDriver(driver Driver, args []string, logger *zap.SugaredLogger) {
 	if len(args) < 2 {
-		ExitWithResult(Fail("Expected at least one argument"))
+		ExitWithResult(Fail(logger, "Expected at least one argument"))
 	}
-
-	log.Printf("'%s %s' called with %s", args[0], args[1], args[2:])
+	logger.With("binary", args[0], "command", args[1], "arguments", args[2:]).Info("Exec called")
```

Comment:
```suggestion
	logger := logger.With("command", args[1])
	logger.With("binary", args[0], "arguments", args[2:]).Debug("Exec called")
```

We shouldn't log the args at info level as they can containe base64 encoded Secrets. Additionally it would be good to pass down the logger into all the interface methods as you originally thought as this way we can have the `"command"` field on the logger for all log messages which will be useful.